app/controllers/callbacks_controller.rb

```diff
@@ -0,0 +1,34 @@
+class CallbacksController < Devise::OmniauthCallbacksController
+    def facebook
+        @user = User.from_omniauth(request.env["omniauth.auth"])
+        sign_in_and_redirect @user
+    end
+ 	
+ 		def google_oauth2
+      # You need to implement the method below in your model (e.g. app/models/user.rb)
+      @user = User.find_for_google_oauth2(request.env["omniauth.auth"], current_user)
+
+      if @user.persisted?
+        flash[:notice] = I18n.t "devise.omniauth_callbacks.success", :kind => "Google"
+        sign_in_and_redirect @user, :event => :authentication
+      else
+        session["devise.google_data"] = request.env["omniauth.auth"]
+        redirect_to new_user_registration_url
+      end
+  	end
+
+  	def self.find_for_google_oauth2(access_token, signed_in_resource=nil)
+	    data = access_token.info
+	    user = User.where(:email => data["email"]).first
```

Comment:
Its rubocop running? This should be sth detected as sth to improve

If you know there is only one you can do this:

```ruby
User.find_by_email:(data['email'])
```